feat: Intercom Read via search

Cobalt0s opened the pull request:
# Changes

* First of all static schema for conversation originally had the right fields then they were replaced with `newsfeed` object. It seems this issue was not noticed as it affects "selectable fields" for Read. The right conversation schema is selected from OpenAPI (coming from POST search conversations)
* Some objects were not properly handled. A better distinction of `Ampersand Object Name` and matching URL path is introduced for exceptions.
![image](https://github.com/user-attachments/assets/212efb13-69bc-42e9-a66b-30d4ccd776dc)
* Tickets was added as a supported object. It can be queried via POST only. For `builders` this means they must pass Since parameter. This is added into [documentation](https://github.com/amp-labs/docs/pull/78).
* Conversations, Contacts, Tickets fallback to search endpoint when Since parameter is provided.
* Activity Logs is now an only GET endpoint that uses `Since` param.

This should complete full support of searchable objects by `updated_at` property.

Cobalt0s commented:
* **#1082** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/1082?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Cobalt0s and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

RajatPawar commented on providers/intercom/metadata/schemas.json:
```diff
@@ -328,6 +339,30 @@
       }
+    },
+    "tickets": {
+      "displayName": "Ticket List",
```
This can be `Tickets` since this seems to be the only way to get all tickets from the API anyway

Cobalt0s replied:
Good catch. I will add a mapping to openapi script

RajatPawar commented on providers/intercom/objectNames.go:
```diff
@@ -6,6 +6,7 @@ import (
 )
 
 // Supported object names can be found under schemas.json.
+// NOTE: tickets can be queried only by using non-empty `Since` parameter.
```
What is the behavior of the tickets search API when we pass in no filters? Would that list all the tickets? If that is the case, I think that's better behavior and would be useful for builders to receive a backfill of all tickets, and then receive only updated tickets. 

i.e. when Since is not present, call `POST /tickets/search` with no request body, else call `POST /tickets/search WHERE updated_at > Since`

Cobalt0s replied:
This is a great hack!
![image](https://github.com/user-attachments/assets/53d7ca6d-21ac-4274-b513-80328dab60ec)
Other objects can be retrieved using GET endpoints. Tickets is the only exception that requires `Search` to see any.

This deep connector has many exceptions now.

Cobalt0s replied:
I have updated the docs removing the NOTE on tickets: https://github.com/amp-labs/docs/pull/78.

Also the full historic read works (temporary replaced "conversations" with "tickets", without Since param):
![image](https://github.com/user-attachments/assets/b911f8ad-6945-4292-8e4c-0ace9d0ae199)

RajatPawar commented on providers/intercom/objectNames.go:
```diff
@@ -18,9 +19,32 @@ var ObjectNameToResponseField = handy.NewDefaultMap(map[string]string{ //nolint:
+	return obj
+})
+
+var incrementalSearchObjectPagination = handy.NewDefaultMap(map[string]int{ //nolint:gochecknoglobals
```
You can add `nolint:gomnd` on the line above the var and it should not lint the var (so that you don't have to repeat it)

Cobalt0s replied:
That worked!

RajatPawar reviewed (changes requested):
Added a comment about how we can list all objects

Cobalt0s commented:
@RajatPawar addressed your comments in the second commit: https://github.com/amp-labs/connectors/pull/1082/commits/1e66d010e2c17f916646493ace3ee7c4ac824bda

RajatPawar commented on providers/intercom/search.go:
```diff
@@ -0,0 +1,116 @@
+		// Perform full read for tickets using POST query.
+		// This is a hack, query is designed to return all objects.
+		return &searchReqPayload{
+			Query: searchQuery{
```
@Cobalt0s this can also be done for a couple more objects, right? Off the top of my head, articles would be one of them

Cobalt0s replied:
Articles search is done via GET and has no time parameter to benefit `Since` filter.

Docs: https://developers.intercom.com/docs/references/rest-api/api.intercom.io/articles/searcharticles
